File: creacion/Abstrac_factory/python/solution/src/test/testMultiIdioma.py
#!/usr/bin/python3
# coding: utf-8

import unittest
from ..main.classes.mensajeEsFactory import MensajesEsFactory
from ..main.classes.mensajeEnFactory import MensajesEnFactory

class TestMultiIdioma(unittest.TestCase):

    def test_es(self):

        # este es el punto clave del abstrac factory, ya que solo cambiando MensajesEsFactory por MensajesEnFactory,
        # cambiamos por completo la famila de clases, por ejemplo pensemos que cuenta Mensajes Idiomas, estamo manejando
        # Bases de datos, como ser Relacional y no relacional, donde las clases Preguntas y Saludos son metodos para obtener
        # de la base de datos

        # que cabrom XD

        factory = MensajesEsFactory()

        preguntas = factory.getPreguntas()

        self.assertEqual('¿qué hora es?', preguntas.preguntaHora())
        self.assertEqual('¿qué tiempo hace?', preguntas.preguntaTiempo())

        saludos = factory.getSaludos()
        self.assertEqual('buenos días', saludos.buenosDias())
        self.assertEqual('buenas tardes', saludos.buenasTardes())

    def test_en(self):

        factory = MensajesEnFactory()

        preguntas = factory.getPreguntas()

        self.assertEqual('what time is it?', preguntas.preguntaHora())
        self.assertEqual('how is the weather?', preguntas.preguntaTiempo())

        saludos = factory.getSaludos()
        self.assertEqual('good morning', saludos.buenosDias())
        self.assertEqual('good afternoon', saludos.buenasTardes())

File: creacion/Abstrac_factory/java/patterns_1_abstract_factory_solucion/src/test/java/org/formacion/abstractfactory/TestMultiIdioma.java
package org.formacion.abstractfactory;

import static org.junit.Assert.*;

import org.junit.Test;

public class TestMultiIdioma {

	@Test
	public void test_es() {
		
		// unico punto de decision
		MensajesAbstractFactory factory = new MensajesEsFactory(); 
		
		Preguntas preguntas = factory.getPreguntas();
		
		assertEquals("¿qué hora es?", preguntas.preguntaHora() );
		assertEquals("¿qué tiempo hace?", preguntas.preguntaTiempo() );
		
		Saludos saludos = factory.getSaludos();
		
		assertEquals("buenos días", saludos.buenosDias());
		assertEquals("buenas tardes", saludos.buenasTardes());
	}
	
	@Test
	public void test_en() {

		// unico punto de decision
		MensajesAbstractFactory factory = new MensajesEnFactory(); 

		Preguntas preguntas = factory.getPreguntas();
		
		assertEquals("what time is it?", preguntas.preguntaHora() );
		assertEquals("how is the weather?", preguntas.preguntaTiempo() );
		
		Saludos saludos = factory.getSaludos();
		
		assertEquals("good morning", saludos.buenosDias());
		assertEquals("good afternoon", saludos.buenasTardes());
	}

}
